src/rigid_body_publisher.cpp

```diff
@@ -65,7 +66,36 @@ namespace utilities
       poseStampedMsg.pose.orientation.w = body.pose.orientation.w;
     }
     return poseStampedMsg;
-  }   
+  }
+  nav_msgs::Odometry getRosOdom(RigidBody const& body, bool newCoordinates)
+  {
+    nav_msgs::Odometry OdometryMsg;
+    if (newCoordinates)
+    {
+      // Motive 1.7+ coordinate system
+      OdometryMsg.pose.pose.position.x = -body.pose.position.x;
+      OdometryMsg.pose.pose.position.y = body.pose.position.z;
+      OdometryMsg.pose.pose.position.z = body.pose.position.y;
+
+      OdometryMsg.pose.pose.orientation.x = -body.pose.orientation.x;
+      OdometryMsg.pose.pose.orientation.y = body.pose.orientation.z;
+      OdometryMsg.pose.pose.orientation.z = body.pose.orientation.y;
```

Comment:
Did you swap these on purpose?

Reply:
Yes y and z axes are swapped in the optitrack coordinate system